CMakeLists.txt

```diff
@@ -69,19 +47,19 @@ ENDIF(Boost_UNIT_TEST_FRAMEWORK_FOUND)
 #
 # This is mainly done avoid name conflicts between files of the interfaces
 # First, add common sources
+set(COMMON_INCLUDE_DIRS "")
+set(library_includes "")
 set(subdirs util backend)
 foreach(subdir ${subdirs})
   aux_source_directory(${CMAKE_CURRENT_SOURCE_DIR}/${subdir}/src library_sources)
   file(GLOB subdir_includes ${CMAKE_CURRENT_SOURCE_DIR}/${subdir}/include/*.h)
-  set(library_includes ${library_includes} ${subdir_includes})
-  #set(SOURCE ${SOURCE} ${sources})
-  #include_directories(${CMAKE_CURRENT_SOURCE_DIR}/${subdir}/include)
-  set(COMMON_INCLUDE_DIRS ${COMMON_INCLUDE_DIRS} ${CMAKE_CURRENT_SOURCE_DIR}/${subdir}/include/)
-  #set(COMMON_INCLUDE_DIRS ${COMMON_INCLUDE_DIRS} ${subdir}/include/)
+  list(APPEND library_includes ${subdir_includes})
+  list(APPEND COMMON_INCLUDE_DIRS ${CMAKE_CURRENT_SOURCE_DIR}/${subdir}/include/)
 endforeach()
 
 add_library(COMMON_OBJECTS OBJECT ${library_sources} ${library_includes})
 target_include_directories(COMMON_OBJECTS PUBLIC ${COMMON_INCLUDE_DIRS})
+set_property(TARGET COMMON_OBJECTS PROPERTY POSITION_INDEPENDENT_CODE ON)
```

Comment:
this looks a bit strange... I would do a static library instead of object library and then expect cmake does the -fPIC option... or do you have reason not to do that?

Reply:
No this was minimal intrusive here, I can check if static library works as well

Reply:
The object libraries were used to avoid issues with the linker optimizing away seemingly unused code

Reply:
Maybe you need -no-as-needed flag? We are using it to prevent the linker from removing seemingly unused libs elsewhere.

Reply:
https://discourse.cmake.org/t/automatically-wrapping-a-static-library-in-whole-archive-no-whole-archive-when-used-during-linking/5883/2

Reply:
As discussed, leave as-is for now and change when removing the mtca4u code